tests/test_cli.py

```diff
@@ -0,0 +1,215 @@
+    :param censys_secret: censys secret value
+    :return: all constant arguments
+    """
+    sk = shodan_key
+    ci = censys_id
+    cs = censys_secret
+    base_args = [
+        "./grinder.py",
+        "-r",
+        "-u",
+        "-cu",
+        "-cp",
+        "-sk",
+        sk,
+        "-ci",
+        ci,
+        "-cs",
+        cs,
+        "-ni",
+        "-d",
+    ]
+    return base_args
+
+
+def test_cli_base_args(cli_args: list) -> None:
+    """
+    Execute base arguments and check results
+    :param cli_args: needed arguments for execution and testing
+    :return: None
+    """
+    p = Popen(["./grinder.py"], stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert "Usage: ./grinder.py -h for help" in str(output)
+
+    p = Popen(cli_args, stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert (
+        "Oops! File with queries was not found. Create it or set name properly."
+        in str(output)
+    )
+    assert "No such file or directory:" in str(output)
+
+
+def test_cli_invalid_base_args() -> None:
+    """
+    Check behavior in case of invalid value following flag that doesn't require value
+    :return: None
+    """
+    base_one = ["./grinder.py", "-r"]
+    flags = ["-r", "-u", "-cu", "-cp", "-ni", "-d", "-nm", "-vs", "-sc", "-vs", "-ts"]
+    invalid_value = "wrong_one"
+    for f in flags:
+        base_one.extend([f, invalid_value])
+        p = Popen(base_one, stdout=PIPE, stderr=PIPE)
+        output, error = p.communicate()
+        assert p.returncode == 1
+        assert "unrecognized arguments:" in str(error)
+
+        base_one.pop()
+        p = Popen(base_one, stdout=PIPE, stderr=PIPE)
+        output, error = p.communicate()
+        assert p.returncode == 1
+        assert (
+            "Oops! File with queries was not found. Create it or set name properly."
+            in str(output)
+        )
+        base_one.pop()
+
+
+def test_cli_invalid_args_after_flags(cli_args: list) -> None:
+    """
+    Check behavior in case of invalid arguments after different flags
+    :param cli_args: base arguments
+    :return: None
+    """
+    invalid_args = cli_args
+    invalid_value = "wrong_one"
+
+    invalid_args.extend(["-q", invalid_value])
+    p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert (
+        "Oops! File with queries was not found. Create it or set name properly."
+        in str(output)
+    )
+    invalid_args[
+        len(invalid_args) - 1
+    ] = "tests/test_data/test_queries/cli_arguments_test.json"
+
+    invalid_args.extend(["-v", invalid_value])
+    p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert "Vendors not found in queries file" in str(output)
+    invalid_args = invalid_args[: len(invalid_args) - 2]
```

Comment:
As I understand this part of code, you want to cut the last 2 elements from list, you can do it in more simpler way:
```python3
In [19]: invalid_args = [1, 2, 3, 4]

In [20]: invalid_args[:-2]
Out[20]: [1, 2]
```